projects/components/src/dropdown/dropdown-focus-handler.service.ts

```diff
@@ -66,28 +72,43 @@ export class DropdownFocusHandlerService implements OnDestroy {
      * automatically moves the focus to first item in the menu on the right side
      */
     listenToArrowKeys(menuContainer: HTMLElement): void {
+        // The following listeners return false when the focus is moved for the key pressed, in order to prevent the default behavior of
+        // that key. For example, to prevent scrolling of page underneath the dropdown when up and down arrow keys are pressed.
         this.unlistenFuncs.push(
             this.renderer.listen(menuContainer, 'keydown.arrowdown', (event: Event) => {
-                this.moveFocus(Direction.DOWN);
                 event.stopPropagation();
+                return !this.moveFocus(Direction.DOWN);
             })
         );
         this.unlistenFuncs.push(
             this.renderer.listen(menuContainer, 'keydown.arrowup', (event: Event) => {
-                this.moveFocus(Direction.UP);
                 event.stopPropagation();
+                return !this.moveFocus(Direction.UP);
             })
         );
         this.unlistenFuncs.push(
             this.renderer.listen(menuContainer, 'keydown.arrowleft', (event: Event) => {
                 if (!this.currentFocusedItem.left) {
-                    return;
+                    return true;
                 }
                 // Close the nested menu before moving focus to left side
-                this.currentFocusedItem.left.closeMenu(event);
-                this.moveFocus(Direction.LEFT);
+                this.currentFocusedItem.left.closeMenu();
                 event.stopPropagation();
+                return !this.moveFocus(Direction.LEFT);
             })
         );
     }
+
+    /**
+     * Whenever escape is pressed on dropdown item, the menu has to be closed and the focus has to be moved to parent menu if any... Just
+     * like when left arrow key is pressed
+     */
+    escapePressed(): void {
+        if (!this.currentFocusedItem.left) {
+            return;
+        }
+        // Close the nested menu before moving focus to left side
+        this.currentFocusedItem.left.closeMenu();
+        this.moveFocus(Direction.LEFT);
```

Comment:
Will `this.moveFocus(Direction.LEFT);` also call `this.currentFocusedItem.left.closeMenu();`?

Reply:
No, movefocus method does not call the close menu